### PR TITLE
KBI0016: Drop local files added in a specific commit

### PR DESCRIPTION
**Pull request by jsheunis**
- Closes https://github.com/psychoinformatics-de/knowledge-base/issues/51

### CODE REVIEW

File: kbi/0016/index.rst [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+   git diff --name-only 42f197501c3293bc1c0c22e36b1618eec706090e..73489f56ecd5eb4dee14c957349f09c0d8b1684d > files_to_drop.out
+
+   file2.txt
+   file3.txt
+
+Note: since we know that the commit with the added files is also the last commit (i.e. it
+corresponds to the current ``HEAD``), we can also omit the second commit hash.
+
+Let's write the filenames into into an output file:
```
Comment (loj) [inline kbi/0016/index.rst:208]:
```suggestion
Let's write the filenames into an output file:
```

**Review by loj — APPROVED** [review]
Just found one typo. Otherwise I think this looks good!

**Comment by jcolomb** [comment]
sorry did not see before. only taking untracked files means that if files were changed, they will not be dropped. (I do not know how many possible status there is...)

This is different in the git diff approach.

That might be what one wants, but one may want to clarify that the text?

**Comment by jsheunis** [comment]
No worries.

I think your concern is already addressed in the text, no?

> The drawback of this approach is that it can't be done after the files have been committed
to ``git`` or ``git-annex`` (i.e. after running ``datalad save``), because then the files'
state would have changed to ``clean``, as with any other previously commited files in the
dataset

**Comment by jcolomb** [comment]
no what I mean is git diff will give you all files that were changed, datalad status + filter for untracked will gives you only files which are new

a file that was modified will not be "untracked" but will be in git diff (?).

**Comment by jcolomb** [comment]
and vice versa, I think the git diff approach would drop the modified files, the text should make the difference more clear.


I think to get the 2 approach as equivalent, filtering for not clean would be the way (but not sure, could not find a list of status `datalad save` can report...)

**Pull request merged**